### ISSUE TITLE
UIShell headerMenu - using esc to close the menu

### ISSUE THREAD

**Issue by jingx-xue** [carbon-design-system/carbon#3661]
## Environment

> Operating system: Win10

> Browser: all browser

> Automated testing tool and ruleset

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using?
latest version
> What did you expect to happen?
should using tab to close the menu like overflow menu compoent
> What happened instead?
esc key is use to close the menu
> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue

1. tab to headerMenu component
2. hit enter to open
3. hit esc to ckose

> Please create a reduced test case in CodeSandbox
> 
> - React:
>  https://codesandbox.io/s/codesandbox-e9eny
## Additional information

- Screenshots or code
- Notes

**Comment by asudoh** [comment]
Saw that ESC key is working with latest `master`.
